packages/components/src/aspect-ratio/component.tsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import { css } from '@emotion/react';
+
+/**
+ * WordPress dependencies
+ */
+import { cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	PolymorphicComponentProps,
+} from '../ui/context';
+import type { AspectRatioProps } from './types';
+import * as styles from './styles';
+import { useCx } from '../utils/hooks';
+import { getValidChildren } from '../ui/utils/get-valid-children';
+
+const { AspectRatioResizer, AspectRatioView } = styles;
+
+function AspectRatio(
+	props: PolymorphicComponentProps< AspectRatioProps, 'div' >,
+	forwardedRef: import('react').Ref< any >
+) {
+	const {
+		children,
+		className,
+		ratio = 1,
+		width,
+		...otherProps
+	} = useContextSystem( props, 'AspectRatio' );
+	const cx = useCx();
+
+	/**
+	 * Noting that only the first valid ReactElement will be actually
+	 * rendered. Other children (if any) are ignored.
+	 */
+	const [ child ] = getValidChildren( children );
```

Comment:
Should we just throw an error if there's more than one child?

Reply:
I considered using [React's `only`](https://reactjs.org/docs/react-api.html#reactchildrenonly). But wasn't sure if we want to error..